backend/app/controllers/uoas.js

```diff
@@ -91,34 +88,95 @@ module.exports = {
         }
 
         co(function* () {
-            var added = yield thunkQuery(
-                'SELECT name, id FROM "UnitOfAnalysis" ' +
+            // Check if Subjects already exist in DB
+            const existingRecords = yield thunkQuery(
+                'SELECT * FROM "UnitOfAnalysis" ' +
                 'WHERE LOWER("UnitOfAnalysis"."name") IN (' + sqlString.toLowerCase() + ') ' +
                 'AND "UnitOfAnalysis"."unitOfAnalysisType" = ' + req.body.unitOfAnalysisType
-
             );
-            var insert = _.difference(uoas, added.map((exist) => exist.name));
-            for (var i = 0; i < insert.length; i++) {
-                var result = yield thunkQuery(UnitOfAnalysis.insert({
-                    name: insert[i],
-                    creatorId: req.user.realmUserId,
-                    ownerId: req.user.realmUserId,
-                    unitOfAnalysisType: req.body.unitOfAnalysisType,
-                    created: new Date(),
-                }).returning(UnitOfAnalysis.id));
-                added.push({name: insert[i], id: _.first(result).id});
+
+            // Empty list to hold inserted or modified records
+            const insertedRecords = [];
+
+            if (_.first(existingRecords)) {
+                for (let i = 0; i < existingRecords.length; i++) {
+                    if (existingRecords.isDeleted !== null) {
+                        const updateObj = {
+                            isDeleted: null
+                        };
+
+                        yield thunkQuery(
+                            UnitOfAnalysis.update(updateObj).where(UnitOfAnalysis.id.equals(existingRecords[i].id))
+                        );
+                    }
+
+                    if (req.body.productId) {
+                        // check that record doesn't already exist in productUOA
+                        const recordInProductUOA = yield thunkQuery(
+                            ProductUOA.select().where(ProductUOA.UOAid.equals(existingRecords[i].id))
+                        );
+
+                        if (!_.first(recordInProductUOA)) { // Record not in productUOA, we can add it
+
+                            yield thunkQuery(ProductUOA.insert({
+                                productId: req.body.productId,
+                                UOAid: existingRecords[i].id,
+                                currentStepId: null,
+                                isComplete: false,
+                            }));
+                        } else if (_.first(recordInProductUOA).isDeleted !== null) {
+
+                            const updateObj = {
+                                isDeleted: null
+                            };
+
+                            const updatedRecordInProductUOA = yield thunkQuery(
```

Comment:
Since it's not reused, you don't need to declare `const updatedRecordInProductUOA` I think.

Reply:
Done!